launcher/tasks/ConcurrentTask.cpp

```diff
@@ -27,10 +27,8 @@ auto ConcurrentTask::getStepTotalProgress() const -> qint64
 
 void ConcurrentTask::addTask(Task::Ptr task)
 {
-    if (!isRunning())
-        m_queue.append(task);
-    else
-        qWarning() << "Tried to add a task to a running concurrent task!";
+    m_queue.append(task);
+    m_total_size += 1;
```

Comment:
I think we might be able to get rid of this and rely solely on `m_queue.size()`.

Or perhaps make it a short method that just returns `m_queue.size()`.

QQueue is just a QList with extras, so there shouldn't be any performance penalty.